als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/ALSConfigurationState.scala

```diff
@@ -110,8 +110,10 @@ case class ALSConfigurationState(editorState: EditorConfigurationState,
     * @param uri
     * @return (name, isScalar)
     */
-  def semanticKeysFor(uri: String): Seq[(String, Boolean)] =
+  def semanticKeysFor(uri: String, excludedDialects: Seq[Dialect] = Seq.empty): Seq[(String, Boolean)] = {
+    val excludedLocations = excludedDialects.map(ed => ed.location().getOrElse(ed.id)) // hack because final id adoption changes reference id
```

Comment:
we have an implicit that does `ed.location().getOrElse(ed.id)` I think (.identifier)